### DISCUSSION TITLE
How to go from a 'pencil' sketch to a filled-in sketch.  See before and after examples.

### DISCUSSION THREAD

**Discussion by ilovetogetspamed** [ImageMagick/ImageMagick#1900] [General]
I'd like to convert  sketch Pronghorn to look like Pronghorn1.  I made Pronghorn 1 in Gimp.  But that would take me about a thousand years.  I have around 3000 files to convert.  I'm sure there's got to be a way with Imagemagick, but it's beyond my knowledge.

To generate Pronghorn1.png I followed the following steps in gimp:
* Selected open paths (e.g. solid lines in face, ears and legs)  with 'Fuzzy Select'
* Colored selected paths white
* Chose Select None from Select Menu
* Selected the closed paths with Select by Color tool (they are transparent areas)
* Bucket Filled selected areas with black
* Exported the file

Ideally I'd like to convert the white open paths back to transparent after filing the closed paths black.

Pronghorn.png

![Pronghorn](https://user-images.githubusercontent.com/3266972/79349196-f88f6d00-7f03-11ea-9372-3c681196dcd3.png)

Pronghorn1.png

![Pronghorn1](https://user-images.githubusercontent.com/3266972/79349199-f9280380-7f03-11ea-9c57-79afa9accc38.png)

**Comment by snibgo** [comment]
What version of IM, on what platform?

IM v7.0.8-64 can't see any paths in your input.

**Reply by ilovetogetspamed** [reply]
The lines in the legs, face and head are what I'm calling 'paths'.  My entire process is convert orignal svg using Gapplin app to png.  tried with ```magick msvg:Pronghorn.svg -colorspace Gray Pronghorn.png``` but it doesn't define the lines very well.  I'm sure it could be improved upon but I am to new to imagemagick to understand all its nuances. 

[Pronghorn.svg.txt](https://github.com/ImageMagick/ImageMagick/files/4482648/Pronghorn.svg.txt)
remome the .txt extention to open svg in imagemagick/inkscape/gimp.

**Comment by ilovetogetspamed** [comment]
Version: ImageMagick 7.0.9-12 Q16 x86_64 2019-12-28 https://imagemagick.org
Copyright: © 1999-2020 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI Modules OpenMP(3.1) 
Delegates (built-in): bzlib freetype heic jng jp2 jpeg lcms ltdl lzma openexr png tiff webp xml zlib

on macOS Mojave version 10.14.6 (18G4032)

**Comment by fmw42** [comment]
To do what you want, you would need to specify a seed point for where you want to flood fill the black area.

Try (Unix syntax)

```
convert animal.png -alpha extract -negate \
-fill black -draw "color 350,400 floodfill" \
\( +clone -negate \) \
-alpha off -compose copy_opacity -composite \
animal_colored.png

```

where 350,400 is the seed point

**Reply by ilovetogetspamed** [reply]
Thanks, one more question is there a quick way to find a seed point along the edge working in from the edge of the picture?

**Reply by fmw42** [reply]
No, I do not see any way to do that.

**Reply by fmw42** [reply]
Your SVG file renders to a completely white png file for me using Imagemagick with Inkscape 9.50

**Reply by urban-warrior** [reply]
We get expected results with inkscape 1.0, the equivalent of Pronghorn.png.

**Reply by ilovetogetspamed** [reply]
Yeah, I know that.  You on a mac? I had trouble too.

using pure Imagemagick you can do it thus:

```
convert msvg:Pronghorn.svg -bordercolor white -border 1 -fill red \
-draw "color 0,0 floodfill" -fill black -opaque white \
-fill white -opaque red -shave 1x1 result.png
```

**Reply by ilovetogetspamed** [reply]
doesn't give the same result as Pronghorn.png but at least you can see the image.  This is all so very new to me.

**Comment by snibgo** [comment]
Editing the SVG makes more sense. For example, change the first "fill:none" to "fill:red", and the second to "fill:yellow". This gives a red-coloured animal with a yellow rear.

If you really want to rasterize the image and then colour that, then first change "stroke-width:0.035" to something more sensible such as "stroke-width:1".
